bi_tools.py

```diff
@@ -1,58 +1,174 @@
+
+    def check_alphabet(self) -> bool:
+        valid_nucleotides = set("ACGTacgt")
+        return all(nucleotide in valid_nucleotides for nucleotide in self.sequence)
+
+
+class RNASequence(NucleicAcidSequence):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+
+    def check_alphabet(self) -> bool:
+        valid_nucleotides = set("ACGUacgu")
+        return all(nucleotide in valid_nucleotides for nucleotide in self.sequence)
+
+
+class AminoAcidSequence(BiologicalSequence):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+
+    def get_length(self) -> int:
+        return len(self.sequence)
+
+    def get_subsequence(self, start: int, end: int) -> str:
+        return self.sequence[start:end+1]
+
+    def print_sequence(self) -> None:
+        print(self.sequence)
+
+    def get_aa_percentage(self) -> None:
+        polar_count = 0
+        nonpolar_count = 0
+
+        polar_acids = set("GAVLIPFW")
+        nonpolar_acids = set("STCMNQYDEKRN")
+
+        polar_count = sum(1 for aa in self.sequence if aa in polar_acids)
+        nonpolar_count = sum(1 for aa in self.sequence if aa in nonpolar_acids)
+
+        print(f"The amino acid sequence has {polar_count/self.get_length()*100}% of polar acids and {nonpolar_count/self.get_length()*100}% of non-polar acids.")
+
+    def check_alphabet(self) -> bool:
+        valid_amino_acids = set("ACDEFGHIKLMNPQRSTVWY")
+        return all(amino_acid in valid_amino_acids for amino_acid in self.sequence)
 
 
 def filter_fastq(input_fastq: str, output_fastq: str,
                  gc_bounds: tuple = (0, 100),
                  length_bounds: tuple = (0, 2**32),
                  quality_threshold: int = 0):
     '''
-    Function filter_fastq, drops fastq-sequences not meeting specified
-    length, gc bound and quality
+    Function filter_fastq, drops fastq-sequences not meeting specified length, gc bound and quality
 
     Args:
-    input_fastq (str)
-    output_fastq (str)
-    gc_bounds (tuple, default = (0, 100))
-    length_bounds (tuple, default = (0, 2**32))
-    quality_threshold (int, default = 0)
+    input_fastq (str): path to the input FASTQ file.
+    output_fastq (str): path to the output FASTQ file.
+    gc_bounds (tuple, default = (0, 100)): lower and upper bounds for GC content.
+    length_bounds (tuple, default = (0, 2**32)): lower and upper bounds for sequence length.
+    quality_threshold (int, default = 0): min quality score.
 
     Returns: None
     '''
-    seqs = fastqf.read_seq_file(input_fastq)
-    for key, value in seqs.items():
-        if ((gc_bounds[0] <= fastqf.get_gc_score(value[0]) <= gc_bounds[1]) and
-                (length_bounds[0] <= len(value[0]) <= length_bounds[1]) and
-                (fastqf.get_quality_score(value[2]) >= quality_threshold)):
+    for record in SeqIO.parse(input_fastq, "fastq"):
+        seq = str(record.seq)
+        gc_content = GC(seq)
```

Comment:
GC устаревший метод у них, лучше 
```suggestion
        gc_content = gc_fraction(seq) * 100
```